### PR TITLE
Finish backup properly after so many retries.

### PR DESCRIPTION
**Pull request by didix21**
Add new option to build libimobiledevice in debug mode

### CODE REVIEW

File: src/idevice/backup2.c [hunk 1]
```diff
@@ -621,12 +622,17 @@ static void do_post_notification(idevice_t device, const char *notification)
     np_client_t np;
     
     lockdownd_client_t lockdown = NULL;
+    lockdownd_error_t lockdown_error;
     
-    if (lockdownd_client_new_with_handshake(device, &lockdown, "idevicebackup2") != LOCKDOWN_E_SUCCESS) {
+    if ((lockdown_error = lockdownd_client_new_with_handshake(device, &lockdown, "idevicebackup2")) != LOCKDOWN_E_SUCCESS) {
+        PRINT_VERBOSE_DEBUG(1, "Could not lockdownd_client_new_with_handshake with lockdown error %d", lockdown_error);
```
Comment (spaque-qustodio) [inline src/idevice/backup2.c:628]: nit: if there's a string representation of the error code, it'd be more useful than a simple number

File: src/idevice/backup2.c [hunk 2]
```diff
@@ -1708,13 +1723,21 @@ void idevice_backup2(struct idevice_backup2_options options, FILE *stream_err, F
...
     
     /* start notification_proxy */
     np_client_t np = NULL;
     ldret = lockdownd_start_service(lockdown, NP_SERVICE_NAME, &service);
+
     if ((ldret == LOCKDOWN_E_SUCCESS) && service && service->port) {
-        np_client_new(device, service, &np);
+        np_error_t np_error = np_client_new(device, service, &np);
+        if(np_error != NP_E_SUCCESS) {
+            PRINT_VERBOSE_DEBUG(1, "ERROR: Could not start service %s, error code %d\n", NP_SERVICE_NAME, np_error);
```
Comment (spaque-qustodio) [inline src/idevice/backup2.c:1736]: Should the error log be something like "Could not create client for service..."?

**Review by spaque-qustodio — APPROVED** [review]
Looks ok, just left comments for a couple of minor things

File: src/idevice/backup2.c (hunk 2 above)
Comment (didix21) [reply src/idevice/backup2.c:1736]: You're right

File: src/idevice/backup2.c (hunk 1 above)
Comment (didix21) [reply src/idevice/backup2.c:628]: You're right but now all the lockdown errors are printed with the error code only, so I've preferred to leave it right this.

**Pull request merged**